addons/base_vat/models/res_partner.py

```diff
@@ -463,21 +463,24 @@ def check_vat_ar(self, vat):
     def default_compact(self, vat):
         return vat
 
-    def _fix_vat_number(self, vat):
+    def _fix_vat_number(self, vat, country_id):
+        code = self.env['res.country'].browse(country_id).code.lower() if country_id else False
         vat_country, vat_number = self._split_vat(vat)
-        check_func_name = 'compact_' + vat_country
+        check_func_name = 'compact_' + code or vat_country
```

Comment:
This still won't work for Monaco. Country code is MC, but they use VAT number such as `FRxxxx`